File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    char[] dict = {'A', 'E', 'I', 'O', 'U'};
    boolean isFound = false;
    
    public int solution(String word) {
        int ans = dfs(word, "");
        
        return ans - 1;
    }
    
    public int dfs(String word, String cur) {
        int n = 1;
        
        if(cur.equals(word)) {
            isFound = true;
            return n;
        }
        
        if(cur.length() == 5) {
            return n;
        }
        
        for(int i = 0; i < 5; i++) {
            n += dfs(word, cur + dict[i]);
            if(isFound) return n;
        }
        
        return n;
    }
}

// dfs

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
ALPHA = "AEIOU"
count = 0

def dfs(current, target):
    global ALPHA
    global count
    
    if current == target:
        return count
    if len(current) == 5:
        return 0
    
    for a in ALPHA:
        count += 1
        answer = dfs(current + a, target)
        if answer:
            return answer

def solution(word):
    
    return dfs("", word)